File: cmd/drift/internal/templates/ios/NativeSwitch.swift
/// NativeSwitch.swift
/// Provides native UISwitch embedded in Drift UI.

import UIKit

// MARK: - Native Switch Container

/// Platform view container for native switch.
class NativeSwitchContainer: NSObject, PlatformViewContainer {
    let viewId: Int
    let view: UIView
    private let switchControl: UISwitch

    init(viewId: Int, params: [String: Any]) {
        self.viewId = viewId
        self.switchControl = UISwitch()
        self.view = switchControl

        super.init()

        // Apply styling
        if let onTint = params["onTintColor"] as? UInt32 {
            switchControl.onTintColor = UIColor(argb: onTint)
        }
        if let thumbTint = params["thumbTintColor"] as? UInt32 {
            switchControl.thumbTintColor = UIColor(argb: thumbTint)
        }

        // Set initial value
        if let value = params["value"] as? Bool {
            switchControl.isOn = value
        }

        // Add target for value changes
        switchControl.addTarget(self, action: #selector(valueChanged), for: .valueChanged)
    }

    func dispose() {
        switchControl.removeTarget(self, action: #selector(valueChanged), for: .valueChanged)
        view.removeFromSuperview()
    }

    @objc private func valueChanged() {
        PlatformChannelManager.shared.sendEvent(
            channel: "drift/platform_views",
            data: [
                "method": "onSwitchChanged",
                "viewId": viewId,
                "value": switchControl.isOn
            ]
        )
    }

    func setValue(_ value: Bool) {
        switchControl.setOn(value, animated: true)
    }

    func updateConfig(_ params: [String: Any]) {
        if let onTint = params["onTintColor"] as? UInt32 {
            switchControl.onTintColor = UIColor(argb: onTint)
        }
        if let thumbTint = params["thumbTintColor"] as? UInt32 {
            switchControl.thumbTintColor = UIColor(argb: thumbTint)
        }
    }
}

File: cmd/drift/internal/templates/android/java/NativeSwitch.kt
/**
 * NativeSwitch.kt
 * Provides native SwitchCompat embedded in Drift UI.
 */
package {{.PackageName}}

import android.content.Context
import android.content.res.ColorStateList
import android.view.View
import android.widget.FrameLayout
import androidx.appcompat.widget.SwitchCompat

/**
 * Platform view container for native switch.
 */
class NativeSwitchContainer(
    context: Context,
    override val viewId: Int,
    params: Map<String, Any?>
) : PlatformViewContainer {

    override val view: View
    private val switch: SwitchCompat

    init {
        switch = SwitchCompat(context).apply {
            // Apply styling
            (params["onTintColor"] as? Number)?.let { color ->
                trackTintList = ColorStateList.valueOf(color.toInt())
            }
            (params["thumbTintColor"] as? Number)?.let { color ->
                thumbTintList = ColorStateList.valueOf(color.toInt())
            }

            // Set initial value
            isChecked = params["value"] as? Boolean ?: false

            // Add listener for value changes
            setOnCheckedChangeListener { _, isChecked ->
                PlatformChannelManager.sendEvent(
                    "drift/platform_views",
                    mapOf(
                        "method" to "onSwitchChanged",
                        "viewId" to viewId,
                        "value" to isChecked
                    )
                )
            }

            layoutParams = FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.WRAP_CONTENT,
                FrameLayout.LayoutParams.WRAP_CONTENT
            )
        }

        view = switch
    }

    override fun dispose() {
        switch.setOnCheckedChangeListener(null)
    }

    fun setValue(value: Boolean) {
        // Temporarily remove listener to avoid feedback loop
        switch.setOnCheckedChangeListener(null)
        switch.isChecked = value
        switch.setOnCheckedChangeListener { _, isChecked ->
            PlatformChannelManager.sendEvent(
                "drift/platform_views",
                mapOf(
                    "method" to "onSwitchChanged",
                    "viewId" to viewId,
                    "value" to isChecked
                )
            )
        }
    }

    fun updateConfig(params: Map<String, Any?>) {
        (params["onTintColor"] as? Number)?.let { color ->
            switch.trackTintList = ColorStateList.valueOf(color.toInt())
        }
        (params["thumbTintColor"] as? Number)?.let { color ->
            switch.thumbTintList = ColorStateList.valueOf(color.toInt())
        }
    }
}
